bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -382,6 +382,13 @@ public void runWithFlags(boolean force, boolean suspendMajor, boolean suspendMin
             lastMinorCompactionTime = System.currentTimeMillis();
             minorCompactionCounter.inc();
         }
+
+        if (force) {
+            if (forceGarbageCollection.compareAndSet(true, false)) {
+                LOG.info("{} Set forceGarbageCollection to true after force GC to make it forceGC-able again.", Thread
```

Comment:
@jiazhai this is "set forcrGarbageCollection to false after force GC", no?

Reply:
Also doesn't https://github.com/apache/bookkeeper/blob/master/bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java#L338 already do that?

Reply:
Thanks. This depends. If `runWithFlags` is called in above safeRun(), this part of change is not needed.
In this PR it is force called by triggerGC, and will not use safeRun.

Reply:
okay